backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/VdsCpuUnit.java

```diff
@@ -18,6 +29,24 @@ public VdsCpuUnit(int socket, int core, int cpu) {
         this.socket = socket;
         this.core = core;
         this.cpu = cpu;
+        this.cpuPinningPolicy = CpuPinningPolicy.NONE;
+        this.vmId = new ArrayList<>();
```

Comment:
you'd probably want to replace it with a concurrent list
I hope we'll be able to minimize the synchronization that is introduced by this PR (e.g., there's no reason to block a thread that handles a vm that goes down on a host while scheduling a vm) and as we minimize it we'll probably find ourselves end up with cases where you iterate a collection while removing an element from that collection

and the name should be vmIds in that case..

Reply:
Maybe we can use Set instead of List as we do not want to have duplicates.